### PR TITLE
Filtered: Upgrade build tools

### LINKED ISSUE #391: Filtered format: Vulnerable dependency "hapijs/hoek"
### Setup
-  SRF version: 2.5.6-alpha / 3.0.0-alpha

### Issue

[hapijs/hoek](https://github.com/hapijs/hoek) as defined in https://github.com/SemanticMediaWiki/SemanticResultFormats/blob/master/formats/filtered/package.json

Known vulnerability found: CVE-2018-3728 - Moderate severity

hoek node module before 5.0.3 or 4.2.1 suffers from a Modification of Assumed-Immutable Data (MAID) vulnerability via...

grafik

grafik

@s7eph4n FYI

### PR DESCRIPTION
**Pull request by s7eph4n**
This PR is made in reference to: #391

This PR upgrade build tools for the Filtered format:
* Gulp v4
* Gulp-CLI v2
* TypeScript v3
* vinyl-source-stream v2

+ 2 minor fixes

This PR includes:
- [ ] Update of RELEASE-NOTES.md
- [ ] Tests (unit/integration)
- [x] CI build passed

Fixes #391

### CODE REVIEW

**Comment by kghbln** [comment]
Thanks a ton!

**Comment by JeroenDeDauw** [comment]
Why bother backporting? Why not just release 3.1?

File: formats/filtered/resources/ts/Filtered/View/MapView.ts [hunk 1]
```diff
@@ -183,7 +183,7 @@ export class MapView extends View {
 
 			// TODO: Limit zoom values to map max zoom
 
-			that.map = L.map( that.getTargetElement().get( 0 ), mapOptions );
+			that.map = L.map( <HTMLElement> that.getTargetElement().get( 0 ), mapOptions );
```
Comment (JeroenDeDauw) [inline formats/filtered/resources/ts/Filtered/View/MapView.ts]: `<HTMLElement>`??? Is this valid JS?

Comment (s7eph4n) [reply formats/filtered/resources/ts/Filtered/View/MapView.ts]: It is a Typescript type assertion.
See https://www.typescriptlang.org/docs/handbook/basic-types.html#type-assertions

**Comment by JeroenDeDauw** [comment]
I suppose it makes some sense for security issues yes. Though given the level of support we can provide it might still be best to just put in a feature release. Not sure if there are even people that pin to 3.0.x rather than 3.x.

**Comment by kghbln** [comment]
Back-ported with 48177cfb6dc7ed4682c6a8139e21e260fd908503

**Pull request merged**